File: fix.py
import sys
import re

with open('app.js', 'r', encoding='utf-8') as f:
    lines = f.readlines()

for i, line in enumerate(lines):
    sq_count = len(re.findall(r"(?<!\\)'", line))
    dq_count = len(re.findall(r'(?<!\\)"', line))
    bq_count = len(re.findall(r"(?<!\\)`", line))
    
    if sq_count % 2 != 0:
        if line.strip().endswith(';'):
            lines[i] = line.rstrip().rstrip(';') + "';\n"
        elif line.strip().endswith(','):
            lines[i] = line.rstrip().rstrip(',') + "',\n"
        else:
            lines[i] = line.rstrip() + "'\n"
            
    elif dq_count % 2 != 0:
        if line.strip().endswith(';'):
            lines[i] = line.rstrip().rstrip(';') + '";\n'
        elif line.strip().endswith(','):
            lines[i] = line.rstrip().rstrip(',') + '",\n'
        else:
            lines[i] = line.rstrip() + '"\n'

    elif bq_count % 2 != 0:
        if line.strip().endswith(';'):
            lines[i] = line.rstrip().rstrip(';') + '`;\n'
        elif line.strip().endswith(','):
            lines[i] = line.rstrip().rstrip(',') + '`,\n'
        else:
            lines[i] = line.rstrip() + '`\n'

with open('app.js', 'w', encoding='utf-8') as f:
    f.writelines(lines)
print('Fixed quotes in app.js')

File: fix.js
const fs = require('fs');
const cp = require('child_process');

let code = fs.readFileSync('app.js', 'utf8');
let lines = code.split('\n');
let maxIterations = 200;

for (let i = 0; i < maxIterations; i++) {
    fs.writeFileSync('app.js', lines.join('\n'));
    try {
        cp.execSync('node -c app.js', { stdio: 'pipe' });
        console.log('Syntax is valid!');
        break;
    } catch (err) {
        let stderr = err.stderr ? err.stderr.toString() : '';
        let match = stderr.match(/app\.js:(\d+)/);
        if (match) {
            let lineNum = parseInt(match[1], 10) - 1;
            let line = lines[lineNum];
            let sqCount = (line.match(/'/g) || []).length;
            let dqCount = (line.match(/"/g) || []).length;
            let bqCount = (line.match(/`/g) || []).length;
            
            if (line.includes('??;')) {
                lines[lineNum] = line.replace('??;', '\'??\';');
            } else if (line.includes('??,')) {
                lines[lineNum] = line.replace('??,', '\'??\',');
            } else if (sqCount % 2 !== 0) {
                if (line.trim().endsWith(';')) lines[lineNum] = line.replace(/;$/, '\';');
                else if (line.trim().endsWith(',')) lines[lineNum] = line.replace(/,$/, '\',');
                else lines[lineNum] = line + '\'';
            } else if (dqCount % 2 !== 0) {
                if (line.trim().endsWith(';')) lines[lineNum] = line.replace(/;$/, '";');
                else if (line.trim().endsWith(',')) lines[lineNum] = line.replace(/,$/, '",');
                else lines[lineNum] = line + '"';
            } else if (bqCount % 2 !== 0) {
                if (line.trim().endsWith(';')) lines[lineNum] = line.replace(/;$/, '`;');
                else if (line.trim().endsWith(',')) lines[lineNum] = line.replace(/,$/, '`,');
                else lines[lineNum] = line + '`';
            } else {
                lines[lineNum] = line.replace('??', '\'??\'');
            }
            console.log('Fixed line ' + (lineNum + 1) + ': ' + lines[lineNum]);
        } else {
            console.log('Could not parse error:', stderr);
            break;
        }
    }
}
